src/ring_theory/polynomial/cyclotomic/eval.lean

```diff
@@ -111,19 +111,42 @@ begin
     { simpa only [eval_X, eval_one, cyclotomic_two, eval_add] using h.right.le } }
 end
 
+lemma cyclotomic_pos_and_nonneg (n : ℕ) {R} [linear_ordered_comm_ring R] (x : R) :
```

Comment:
Is there a reason why this lemma is stated using `∧`? I think splitting it is a good idea.

Reply:
That's exactly what is done below. It's a way to deduplicate the proofs of those two lemmas.

Reply:
```suggestion
/-- Auxiliary lemma to deduplicate the proofs of both halves. See `cyclotomic_pos'` and `cyclotomic_nonneg`. -/
lemma cyclotomic_pos_and_nonneg (n : ℕ) {R} [linear_ordered_comm_ring R] (x : R) :
```

Reply:
> That's exactly what is done below. It's a way to deduplicate the proofs of those two lemmas.

Nice trick!